virttest/qemu_storage.py

```diff
@@ -26,21 +26,78 @@
                                    decode_to_text)
 
 
+def filename_to_file_opts(filename):
+    """Convert filename into file opts, used by both qemu-img and qemu-kvm"""
+    file_opts = {}
+    if not filename:
+        file_opts = {}
+    elif filename.startswith('iscsi:'):
+        filename_pattern = re.compile(
+            r'iscsi://(?P<auth_portal>.+?)/(?P<target>.+?)/(?P<lun>\d+)')
+        matches = filename_pattern.match(filename)
+        if matches:
+            file_opts = {'driver': 'iscsi',
+                         'transport': 'tcp',
+                         'target': matches.group('target'),
+                         'lun': matches.group('lun')}
+
+            auth_portal = matches.group('auth_portal')
+            auth_portal_pattern = re.compile(
+                r'(?P<user>.+?):(?P<password>.+?)@(?P<portal>.+)')
+            auth_portal_matches = auth_portal_pattern.match(auth_portal)
+            if auth_portal_matches:
+                file_opts['user'] = auth_portal_matches.group('user')
+                file_opts['portal'] = auth_portal_matches.group('portal')
+            else:
+                file_opts['portal'] = auth_portal
```

Comment:
With using `?` ([reference](https://docs.python.org/3.7/library/re.html?highlight=re#index-5)), the two conditions can be handled by the same one pattern.

Reply:
Done